hub/core/dataset/dataset.py

```diff
@@ -1452,8 +1452,23 @@ def _set_derived_attributes(self, verbose: bool = True):
                 self.index = Index.from_json(self.meta.default_index)
         elif not self._read_only:
             self._lock()  # for ref counting
+
         if not self.is_iteration:
-            self.index.validate(self.num_samples)
+            group_index = self.group_index
+            max_tensor_length = max(
+                map(
```

Comment:
I would assign the results of `map` and `filter` to variables with helpful names.